src/cli/utils.ts

```diff
@@ -0,0 +1,127 @@
+import fs from "fs";
+
+import globTool from "glob-promise";
+
+type KeyValuePair = { [key: string]: string };
+
+interface Config {
+  [key: string]: {
+    element: string;
+    css: string;
+    variants: {
+      [key: string]: KeyValuePair;
+    };
+    compoundVariants: KeyValuePair[];
+  };
+}
+
+export const extractStyles = (path: string) => {
+  const buffer = fs.readFileSync(path);
+  const fileContent = buffer.toString();
+  return fileContent.match(
+    /([a-zA-Z_]*)(?:[.]{1})([a-zA-Z_]+[\w\-_]*)(?:[\s\.\,\{\>#\:]{0})/gim
+  );
+};
+
+export const stylesToConfig = (styles: string[]) => {
+  const config: Config = {};
+  styles.forEach((item) => {
+    const parts = item.split(".");
+    const element = parts[0];
+    const className = parts[1];
+    const chunks = className.split("_");
+
+    if (chunks.length === 2) return;
+
+    if (chunks.length >= 1) {
+      const component = chunks[0];
+
+      if (!config[component]) {
+        config[component] = {
+          variants: {},
+          compoundVariants: [],
+          css: component,
+          element,
+        };
+      }
+
+      if (chunks.length === 3 || chunks.length === 4) {
+        const variant = chunks[1];
+        const option = chunks[2];
+        if (!config[component].variants[variant]) {
+          config[component].variants[variant] = {};
+        }
+        config[component].variants[variant][option] = className;
+      } else if (chunks.length > 4) {
+        const variants = chunks.slice(1, chunks.length);
+
+        const vars = variants.reduce((acc, cur, i, arr) => {
+          if (i % 2 !== 0 || i + 1 >= arr.length) return acc;
+          acc[cur] = arr[i + 1];
+          return acc;
+        }, {} as KeyValuePair);
+
+        config[component].compoundVariants.push({
+          ...vars,
+          css: className,
+        });
+      }
+    }
+  });
+  return config;
+};
+
+export const generateOutput = (config: Config, cssFilename: string) => {
+  let s = "";
+  s += `import { styled } from "@phntms/css-components";\n\n`;
+  s += `import css from "./${cssFilename}";\n\n`;
```

Comment:
Might be missing something obvious, but why is there double `\n\n` here but only single `\n` in the logic below?

Reply:
So we have a space between the imports.